test-site/src/components/MapPin.tsx

```diff
@@ -1,9 +1,48 @@
-import { PinComponent } from '@yext/search-ui-react';
+import { useEffect, useRef, useState } from 'react';
+import ReactDOM from 'react-dom';
+import { Map, Popup, LngLatLike } from 'mapbox-gl';
+import { PinComponent, Coordinate } from '@yext/search-ui-react';
+import { Result } from '@yext/search-headless-react';
 import { Location } from '../pages/LocationsPage';
+import { MapPopup } from './MapPopup';
+
+interface MapPinProps {
+  result: Result<Location>
+  index: number
+  mapbox: Map
+}
+
+const transformToMapboxCoord = (coordinate: Coordinate): LngLatLike => ({
+  lng: coordinate.longitude,
+  lat: coordinate.latitude,
+});
+
+export const MapPin: PinComponent<Location> = (props: MapPinProps) => {
+  const popupRef = useRef(new Popup({ offset: 15 }));
+  const initialRender = useRef(true);
+  const [active, setActive] = useState(false);
+
+  useEffect(() => {
+    if (!initialRender.current && active) {
+      const popup = document.createElement('div');
+      ReactDOM.render(<MapPopup entity={props.result} />, popup);
+      popupRef.current
+        .setLngLat(coordinate)
+        .setDOMContent(popup)
+        .addTo(props.mapbox);
+    }
+    initialRender.current = false;
+  }, [active]);
```

Comment:
I think this effect is missing dependencies? It looks like there's a number of eslint complaints, TIL our linting workflow doesn't run when you merge into a feature branch. We probably will update this in the future but in the meantime if you could manually run eslint?